### DISCUSSION TITLE
unplugin-fonts fails to inject into <head>

### DISCUSSION THREAD

**Discussion by Maytha8** [vikejs/vike#2032] [Discussion]
Hi! I'm trying to use `unplugin-fonts` with Vike, configured to use Google Fonts as its source, so it can inject the `<link>` tags pointing to Google Fonts into `<head>`, but it fails to do that. It works when using local (custom) fonts because it can update it's "unfonts.css" stylesheet. A stripped down reproduction can be found [https://github.com/Maytha8/vike-fonts-repro](https://github.com/Maytha8/vike-fonts-repro). 

**Expected output:**
![image](https://github.com/user-attachments/assets/21ae8c04-b9a5-4ac1-81e1-d4cab8b9033a)

**Actual output:**
![image](https://github.com/user-attachments/assets/51a388ff-1ae3-4637-8802-b96c45f74f67)

Is there some sort of configuration option with either Vike or perhaps the plugin that can work around this, or is this a deeper problem? Thanks.

**Comment by brillout** [comment]
> it fails to do that

Can you elaborate on that?

Possibly related: https://github.com/vikejs/vike/issues/224.

**Reply by Maytha8** [reply]
> Can you elaborate on that?

Nothing gets injected into the `<head>` when it should be injecting a `<link>` to the Google Fonts stylesheet.

**Reply by brillout** [reply]
See https://github.com/vikejs/vike/issues/224. Does unplugin-fonts export a function that can be used on the server to transform the HTML upon SSR?

**Reply by brillout** [reply]
Keep us updated if you find a solution, so that we can add it to the list of workarounds.
